pkg/table/tables/tables.go

```diff
@@ -1403,6 +1399,21 @@ func (t *TableCommon) RemoveRecord(ctx table.MutateContext, h kv.Handle, r []typ
 	return err
 }
 
+func projectPublicColData(t *TableCommon, deletableData []types.Datum) (publicData []types.Datum, colIDs []int64) {
+	publicColLen := len(t.Cols())
+	publicData = make([]types.Datum, 0, publicColLen)
+	colIDs = make([]int64, 0, publicColLen+1)
+	deletableCols := t.DeletableCols()
+	for i, d := range deletableData {
+		dCol := deletableCols[i]
+		if dCol.State == model.StatePublic {
```

Comment:
For lossy modifying a column, it consists of two steps:

1. add a hidden column
2. drop the old column

Since DDL is only synced when the job state is done, the hidden column does not exists in downstream.